templates/nvidia-rag-canonical/README.md

```diff
@@ -0,0 +1,121 @@
+
+# nvidia-rag-canonical
+
+This template performs RAG using Milvus Vector Store and NVIDIA Models (Embedding and Chat).
+
+## Environment Setup
+
+You should export your NVIDIA API Key as an environment variable.
+If you do not have an NVIDIA API Key, you can create one by following these steps:
+1. Create a free account with the [NVIDIA GPU Cloud](https://catalog.ngc.nvidia.com/) service, which hosts AI solution catalogs, containers, models, etc.
+2. Navigate to `Catalog > AI Foundation Models > (Model with API endpoint)`.
+3. Select the `API` option and click `Generate Key`.
+4. Save the generated key as `NVIDIA_API_KEY`. From there, you should have access to the endpoints.
+
+```shell
+export NVIDIA_API_KEY=...
+```
+
+For instructions on hosting the Milvus Vector Store, refer to the section at the bottom.
+
+## Usage
+
+To use this package, you should first have the LangChain CLI installed:
+
+```shell
+pip install -U langchain-cli
+```
+
+To use the NVIDIA models, install the Langchain NVIDIA AI Endpoints package:
+```shell
+pip install -U langchain_nvidia_aiplay
+```
+
+To create a new LangChain project and install this as the only package, you can do:
+
+```shell
+langchain app new my-app --package nvidia-rag-canonical
+```
+
+If you want to add this to an existing project, you can just run:
+
+```shell
+langchain app add nvidia-rag-canonical
+```
+
+And add the following code to your `server.py` file:
+```python
+from nvidia_rag_canonical import chain as rag_nvidia_chain
+
+add_routes(app, rag_nvidia_chain, path="/nvidia-rag")
+```
+
+If you want to set up an ingestion pipeline, you can add the following code to your `server.py` file:
+```python
+from rag_nvidia_canonical import ingest as rag_nvidia_ingest
+
+add_routes(app, rag_nvidia_ingest, path="/nvidia-rag-ingest")
+```
+Note that for files ingested by the ingestion API, the server will need to be restarted for the newly ingested files to be accessible by the retriever.
```

Comment:
Should we also provide steps to delete any existing ingested files?

Reply:
I don't think this is needed. While we do offer an ingestion API, it seems from other RAG templates that the expectation is that developers independently manage their vectorstore. The chain is meant more for users.